include/wil/registry.h

```diff
@@ -17,16 +17,3139 @@
+        }
+
+#if defined(__SDDL_H__)
+        /**
+         * \brief Creates a new HKEY to the specified path
+         *        see RegCreateKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param path A string to append to the HKEY to attempt to open
+         *        can be nullptr if not needed
+         * \param security_descriptor A string-format security descriptor specifying the security descriptor for the new HKEY
+         *        see ConvertStringSecurityDescriptorToSecurityDescriptorW
+         * \param[out] hkey A pointer to an HKEY parameter to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure
+         *         does not throw C++ exceptions
+         *
+         * \remark Because security functions and types are referenced, the <sddl.h> header must be previously included
+         */
+        inline HRESULT create_key_nothrow(HKEY key, PCWSTR path, _In_opt_ PCWSTR security_descriptor, _Out_ HKEY* hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT
+        {
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.create_key(path, security_descriptor, hkey, access);
+        }
+#endif // #if defined(__SDDL_H__)
+
+
+#if defined(WIL_ENABLE_EXCEPTIONS)
+        /**
+         * \brief Queries for number of sub-keys
+         * \param key The HKEY to query for number of sub-keys
+         * \return The queried number of sub-keys if succeeded
+         * \exception std::exception will be thrown on failure (including wil::ResultException)
+         */
+        inline size_t get_child_key_count(HKEY key)
+        {
+            DWORD numSubKeys{};
+            THROW_IF_WIN32_ERROR(::RegQueryInfoKeyW(
+                key, nullptr, nullptr, nullptr, &numSubKeys, nullptr,
+                nullptr, nullptr, nullptr, nullptr, nullptr, nullptr));
+            return numSubKeys;
+        }
+
+        /**
+         * \brief Queries for number of values
+         * \param key The HKEY to query for number of values
+         * \return The queried number of value if succeeded
+         * \exception std::exception will be thrown on failure (including wil::ResultException)
+         */
+        inline size_t get_child_value_count(HKEY key)
+        {
+            DWORD numSubValues{};
+            THROW_IF_WIN32_ERROR(::RegQueryInfoKeyW(
+                key, nullptr, nullptr, nullptr, nullptr, nullptr,
+                nullptr, &numSubValues, nullptr, nullptr, nullptr, nullptr));
+            return numSubValues;
+        }
+#endif // #if defined(WIL_ENABLE_EXCEPTIONS)
+        /**
+         * \brief Queries for number of sub-keys
+         * \param key The HKEY to query for number of sub-keys
+         * \param[out] numSubKeys A pointer to a DWORD to receive the returned count

```

Comment:
Nit: for variables like this could we use something like `out pointer to returned count`. Type is explicit in the signature.

Reply:
sorry, I'm confused. numSubKeys is an out pointer to the returned count. it's an _Out_ DWORD*

Reply:
Oh, `outptr DWORD to returned count` if you'd like. The current wording struck me as a little wordy.

Reply:
Just did an audit, the current wording is clear enough IMO.